errbot/backends/slack.py

```diff
@@ -394,17 +417,21 @@ def _message_event_handler(self, event):
 
     def userid_to_username(self, id_):
         """Convert a Slack user ID to their user name"""
-        user = [user for user in self.sc.server.users if user.id == id_]
+        users = self.api_call('users.list')
```

Comment:
I figured out the main differences:
`api_call('users.list')` can be called without having the bot connected via rtm.
`sc.server.users` only has values after `sc.rtm_connect()` is executed.

Because I was altering the bot_prefixes beofre `sc.rtm_connect()` was executed, I had to use `api_call()` to get a list of users.

To preserve the original method calls, I simply moved the altering of bot_alternatives to just after `sc.rtm_connect()` is called.